xdsl/dialects/csl/csl_wrapper.py

```diff
@@ -0,0 +1,308 @@
+                )
+
+    @classmethod
+    def parse(cls, parser: Parser):
+        args = parser.parse_op_args_list()
+        operands = parser.resolve_operands(args, [], parser.pos)
+
+        props = parser.parse_optional_properties_dict()
+        props_l = list(props.items())
+        assert len(props_l) >= 2
+        name, width = props_l.pop(0)
+        assert name == "width"
+        name, height = props_l.pop(0)
+        assert name == "height"
+        params = dict[str, Attribute]()
+        for name, value in props_l:
+            params[name] = value
+
+        parser.parse_punctuation("(")
+        layout_module = parser.parse_region()
+        parser.parse_punctuation(",")
+        program_module = parser.parse_region()
+        parser.parse_punctuation(")")
+
+        return cls(
+            operands=operands,
+            result_types=[],
+            regions=[layout_module, program_module],
+            properties={
+                "width": width,
+                "height": height,
+                "params": DictionaryAttr(data=params),
+            },
+            attributes={},
+        )
+
+    def print(self, printer: Printer):
+        printer.print("() ")
+
+        params: dict[str, Attribute] = {"width": self.width, "height": self.height}
+        params.update(self.params.data)
+
+        printer.print("<")
+        printer.print_attr_dict(params)
+        printer.print("> (")
+        printer.print_region(self.layout_module, print_entry_block_args=True)
+        printer.print(", ")
+        printer.print_region(self.program_module, print_entry_block_args=True)
+        printer.print(")")
+
+    def get_layout_arg(self, name: str):
+        """Retrieve layout block arg for name that is x, y, or one of the properties"""
+        available_arg_names = ["x", "y", "width", "height"] + list(
+            self.params.data.keys()
+        )
+        assert (
+            name in available_arg_names
+        ), f"{name} does not refer to a block arg of this layout_module"
+        idx = available_arg_names.index(name)
+        return self.layout_module.block.args[idx]
+
+    def get_program_arg(self, name: str):
+        """Retrieve program block arg for name that is one of the properties or a param set up by layout yield"""
+        layout_yield = self.layout_module.block.last_op
+        assert isinstance(layout_yield, YieldOp)
+        yield_args = (
+            []
+            if layout_yield.fields is None
+            else [y.data for y in layout_yield.fields.data]
+        )
+        available_arg_names = (
+            ["width", "height"] + list(self.params.data.keys()) + yield_args
+        )
+        assert (
+            name in available_arg_names
+        ), f"{name} does not refer to a block arg of this program_module"
+        idx = available_arg_names.index(name)
+        return self.program_module.block.args[idx]
```

Comment:
I would really like to see pytest tests for helpers like these (and type annotations).

Reply:
Added type annotations

Reply:
We still need pytests

Reply:
we now have pytests